Backend/app/models/room_booking.rb

```diff
@@ -4,5 +4,23 @@ class RoomBooking < ApplicationRecord
   has_one :event
   validates_associated :user, :room, :event
   validates :user_id, :room_id, :starts_at, :ends_at, presence: true 
+    validate :end_date_after_start_date
+
+  private
+  def end_date_after_start_date
+    return if ends_at.blank? || starts_at.blank?
+
+    if ends_at < starts_at
+      errors.add(:ends_at, "must be after the start date")
+    end
+ end
+
+   def starting_time
```

Comment:
fix the indentation, and why do we need this?

Reply:
to convert the time and check if (starting_time<ending_time)

Reply:
Why when do we need that conversion? Is it something that can be done in serializer? You can also configure Rails to format all of time and dates automatically.